### PR TITLE
build(*): update the module to rollup

### PR DESCRIPTION
**Pull request by euhmeuh**
Enjoy.

### CODE REVIEW

**Comment by FredericEspiau** [comment]
Could you please be a bit more thorough @euhmeuh  ? What did you actually do, the commits and description of you PR are terse to say the least :)

File: rollup.config.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+const config = configGenerator({
+  input: './src/index.js',
+});
+
+export default [
+  config.cjs(),
+  config.umd({
+    output: {
+      globals: {
+        '@ovh-ux/translate-async-loader': 'translate-async-loader',
```
Comment (antleblanc) [inline rollup.config.js:12]: Could you please cleanup these globals?

File: src/controller.js [hunk 2]
```diff
@@ -0,0 +1,219 @@
+import webNotification from 'simple-web-notification';
+
+class ChatbotCtrl {
+  constructor(
```
Comment (antleblanc) [inline src/controller.js:4]:
```suggestion
  /* @ngInject */
  constructor(
```

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+const moduleName = 'ngOvhChatbot';
+
+angular
+  .module(moduleName, [
+    'ngAnimate',
+    'ngEmbed',
+    'ovh-angular-user-pref',
+  ])
+  .controller('chatbotController', ChatbotCtrl)
```
Comment (antleblanc) [inline src/index.js:20]:
```suggestion
  .controller('ChatbotController', ChatbotCtrl)
```

File: src/service.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    items: [],
+    buttons: [],
+  },
+];
+
+class ChatbotServiceProvider {
+  constructor() {
+    const self = this;
+    this.chatbotUrl = '/engine/2api/chatbot';
+    this.$get = ['$http', '$q', ($http, $q) => {
```
Comment (antleblanc) [inline src/service.js:14]:
```suggestion
    this.$get = /* @ngInject */ ($http, $q) => {
```

File: src/service.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
+const fakeChatbotData = [
```
Comment (antleblanc) [inline src/service.js:1]: Shouldn't be removed?

File: src/index.js [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+  .provider('chatbotService', ChatbotServiceProvider)
+  .directive('ovhChatbot', directive)
+  .run(($templateCache) => {
+    $templateCache.put('ng-ovh-chatbot-icons-template.html', iconsTemplate);
+    $templateCache.put('ng-ovh-chatbot-embedded-template.html', embeddedTemplate);
+  })
+  .constant('CHATBOT_MESSAGE_TYPES', {
```
Comment (antleblanc) [inline src/index.js:27]: Both `CHATBOT_MESSAGE_TYPES` and `CHATBOT_NGEMBED_OPTIONS` could be extracted to a `constants.js` file.

File: src/directive.js [hunk 7]
```diff
@@ -0,0 +1,33 @@
+import template from './templates/index.html';
+
+export default () => ({
+  template,
+  restrict: 'E',
+  replace: true,
+  controller: 'chatbotController',
```
Comment (antleblanc) [inline src/directive.js:7]:
```suggestion
  controller: 'ChatbotController',
```

File: src/directive.js [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+  controllerAs: 'ctrl',
```
Comment (antleblanc) [inline src/directive.js:8]:
```suggestion
  controllerAs: '$ctrl',
```

File: package.json [hunk 9]
```diff
@@ -1,68 +1,60 @@
...
-    "karma-spec-reporter": "0.0.31",
-    "less-plugin-remcalc": "0.0.1",
-    "matchdep": "^0.3.0",
-    "phantomjs": "^2.1.7"
-  },
-  "config": {
-    "commitizen": {
-      "path": "./node_modules/cz-ovh-commit"
-    }
+    "ovh-angular-user-pref": "^0.3.1"
```
Comment (antleblanc) [inline package.json:58]: Could you try the latest prerelease of `ng-ovh-user-pref` instead? 

```sh
yarn remove ovh-angular-user-pref 
yarn add --peer @ovh-ux/ng-ovh-user-pref
```

ref: https://github.com/ovh-ux/ng-ovh-user-pref/tree/v1.0.0-beta.0

Thanks:)

**Review by antleblanc — CHANGES_REQUESTED** [review]
Well done.

I just add some comments regarding that some content has been copy/pasted and could be refactored I guess.

File: src/index.less [hunk 10]
```diff
@@ -1,4 +1,4 @@
-@import (optional, reference) "ovh-ui-kit/packages/oui/stylekit";
+@import (optional, reference) "../node_modules/ovh-ui-kit/packages/oui-color/_variables";
```
Comment (marie-j) [inline src/index.less:1]: Would it make sense to use `~` instead ?

**Comment by euhmeuh** [comment]
Comments fixed in https://github.com/ovh-ux/ng-ovh-chatbot/pull/8. Closing.

**Pull request closed without merging**